pkg/netstate/netstate.go

```diff
@@ -39,10 +41,23 @@ type DB interface {
 	Delete([]byte) error
 }
 
+func validateAuth(xAPIKeyBytes []byte) error {
+	if !auth.ValidateAPIKey(string(xAPIKeyBytes)) {
+		return grpc.Errorf(codes.Unauthenticated, "Invalid API credential")
+	}
+	return nil
+}
+
 // Put formats and hands off a file path to be saved to boltdb
 func (s *Server) Put(ctx context.Context, putReq *pb.PutRequest) (*pb.PutResponse, error) {
 	s.logger.Debug("entering netstate put")
 
+	xAPIKeyBytes := []byte(putReq.XApiKey)
+	if err := validateAuth(xAPIKeyBytes); err != nil {
+		s.logger.Error("unauthorized request")
+		return &pb.PutResponse{}, nil
```

Comment:
should you return the error ?

Reply:
yeah, i get the runtime error- if i return nil, err

Reply:
not nil, err but `return &pb.PutResponse{}, err`

Reply:
helped in slack, but yeah, this should be `return nil, err`